bugz.js

```diff
@@ -110,50 +110,55 @@ class BugzillaBug extends Bug {
 }
 
 async function loadIssuesFromGithubRepo(searchParams) {
-  let {search, filters} = searchParams;
-
-  let projectIssues = gh.getIssues(search.user, search.project);
-  let queryParams = {
-    state: (filters && filters.open) ? "open" : "closed",
-  };
-  let response = await projectIssues.listIssues(queryParams);
+  try {
+    let {search, filters} = searchParams;
 
-  let mapped = response.data.map(is => {
-    let data = {
-      id: "gh:" + is.id,
-      assignee: null,
-      points: null,
-      title: is.title,
-      lastChangeDate: is.updated_at,
-      url: is.html_url,
-      whiteboard: null,
-      priority: null,
-      labels: null,
-      project: search.project,
-      isPullRequest: ("pull_request" in is),
+    let projectIssues = gh.getIssues(search.user, search.project);
+    let queryParams = {
+      state: (filters && filters.open) ? "open" : "closed",
     };
+    let response = await projectIssues.listIssues(queryParams);
+
+    let mapped = response.data.map(is => {
+      let data = {
+        id: "gh:" + is.id,
+        assignee: null,
+        points: null,
+        title: is.title,
+        lastChangeDate: is.updated_at,
+        url: is.html_url,
+        whiteboard: null,
+        priority: null,
+        labels: null,
+        project: search.project,
+        isPullRequest: ("pull_request" in is),
+      };
+
+      if (is.assignee) {
+        data.assignee = is.assignee.login;
+      } else if (data.isPullRequest) {
+        data.assignee = is.user.login;
+      }
 
-    if (is.assignee) {
-      data.assignee = is.assignee.login;
-    } else if (data.isPullRequest) {
-      data.assignee = is.user.login;
-    }
-
-    let labelNames = is.labels.map(l => l.name);
-    data.labels = labelNames;
-    if (data.isPullRequest) {
-      data.labels.push("pr");
-    }
+      let labelNames = is.labels.map(l => l.name);
+      data.labels = labelNames;
+      if (data.isPullRequest) {
+        data.labels.push("pr");
+      }
 
-    let priorityLabel = labelNames.find(l => l.match(/^priority:[0-9]$/));
-    if (priorityLabel) {
-      data.priority = priorityLabel.split(":")[1];
-    }
+      let priorityLabel = labelNames.find(l => l.match(/^priority:[0-9]$/));
+      if (priorityLabel) {
+        data.priority = priorityLabel.split(":")[1];
+      }
 
-    return new GithubIssue(data);
-  });
+      return new GithubIssue(data);
+    });
 
-  return mapped;
+    return mapped;
+  } catch (e) {
+    // A failure in loading GitHub issues shouldn't break the site.
+    return [];
```

Comment:
Can we print a line of log here? So that we don't get crazy if this happens and we forget :)

Reply:
hah, my dev tools logs the line that the XHR request failed, but I think that's not standard. Let me add something.